### ISSUE TITLE
Direct std::chrono::duration support ?

### ISSUE THREAD

**Issue by dfleury2** [msgpack/msgpack-c#931]
HI,
I have seen the work done for std::chrono::<clock_type>::time_point

But I wonder how can it be extended to support std::chrono::duration directly ?
I have not seen if it was not done yet. 
Something like:

```cpp
#include <msgpack.hpp>
#include <chrono>

int main() {
    std::chrono::nanoseconds ns{650};
    msgpack::sbuffer sbuf;  // simple buffer
    msgpack::pack(&sbuf, ns);
}
```

Regards,

**Comment by redboltz** [comment]
It seems that no duration direct support implemented yet.

**Comment by redboltz** [comment]
I tried to implement `std::chrono::duration` adaptor. But I got a problem.
duration has template parameters.
https://en.cppreference.com/w/cpp/chrono/duration

I wrote the following testing code.
https://wandbox.org/permlink/sUw9nx6gvohwZXHt

I guess the value that is returned by `count()` is a candidate for packing.
However, it depends on Num and Den. In addition Rep need to choose carefully to avoid integer overflow.

For example, always pack as nanoseconds is a bad idea. User might treat some years.

MessagePack could be used as inter programming language communication.
Finally, I got a conclusion that there is no appropriate MessagePack type that supports all kinds of `std::chrono::duration`.
It is application domain.
So msgpack-c can't provide the adaptor.
